lib/env_args.h

```diff
@@ -1,16 +1,53 @@
 // env_args.h is a 1 file library to get arguments from env.
-// TODO: Refactor this so that we could parse argv too
+#include <stdbool.h>
 #include <stdlib.h>
 
 #ifndef CLWB_ENV_ARGS_H
 #define CLWB_ENV_ARGS_H
 
-#define env_arg_str(name, out, default) \
-        char *env = getenv(name); \
-        if (env != NULL) { \
-            out = env; \
-        } else { \
-            out = default; \
-        }
+#define env_arg_str(name, out, default)                                        \
+    char* env = getenv(name);                                                  \
+    if (env != NULL) {                                                         \
+        out = env;                                                             \
+    } else {                                                                   \
+        out = default;                                                         \
+    }
 
-#endif //CLWB_ENV_ARGS_H
+#define export_opt_c_str(opt_name) char* get_##opt_name();
+#define export_opt_bool(opt_name) bool get_##opt_name();
+#define export_opt_enum(opt_name, opt_type) opt_type get_##opt_name();
+#define export_opt_bits_set(opt_name, opt_smallname)                           \
+    bool get_##opt_name_##opt_smallname();
+
+void
+parse_commandline_options();
+
+enum FOptimizeOptions
+{
+    opt_dce,
+    opt_instsimplify,
+    opt_inlining,
+    opt_strip
+};
+
+enum OLevel
+{
+    g,
```

Comment:
Replace `g` with `None` or `DebugInfo` might be better? Use `g` in the argument but internal `g` is probably a bad idea